trade-finance-manager-api/src/v1/helpers/amendment.helpers.js

```diff
@@ -310,24 +312,50 @@ const firstTaskEmailConfirmation = async (facilityId, amendmentId, auditDetails)
   await api.updateFacilityAmendment(facilityId, amendmentId, payload, auditDetails);
 };
 
-// sends email for first amendment task when pim submit amendment
-const sendFirstTaskEmail = async (taskVariables, auditDetails) => {
-  const { amendment, dealSnapshot, facilityId, amendmentId } = taskVariables;
+/**
+ * Sends an email notification for the first task of an amendment.
+ *
+ * This function performs the following operations:
+ * 1. Extracts necessary details from the deal and amendment.
+ * 2. Retrieves the first task from the amendment tasks.
+ * 3. Constructs the email variables and sends the email using the TFM email service.
+ * 4. Updates the flag to indicate that the email has been sent if the email is successfully sent.
+ *
+ * @param {Object} deal - The deal object containing deal and amendment details.
+ * @param {Object} deal.amendment - The amendment details.
+ * @param {Object} deal.dealSnapshot - The snapshot of the deal details.
+ * @param {string} deal.facilityId - The ID of the facility.
+ * @param {string} deal.amendmentId - The ID of the amendment.
+ * @param {Object} deal.dealSnapshot.exporter - The exporter details.
+ * @param {string} deal.dealSnapshot.exporter.companyName - The name of the exporter company.
+ * @param {string} [deal.dealSnapshot.ukefDealId] - The UKEF deal ID for GEF deals.
+ * @param {Object} [deal.dealSnapshot.details] - The details object for BSS deals.
+ * @param {string} [deal.dealSnapshot.details.ukefDealId] - The UKEF deal ID for BSS deals.
+ * @param {Object} auditDetails - The audit details for logging purposes.
+ * @returns {Promise<void>} - A promise that resolves when the email has been sent and the flag has been updated.
+ * @throws {Error} - Logs the error if any error occurs during the email sending process.
```

Comment:
I think this function never throws an error that isn't caught and then silently ignored (well, technically `getFirstTask` could throw and that would not be caught by the `catch` block)?